### PR TITLE
Drop filter routes

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

File: app/models/endpoint.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ class Endpoint < ApplicationRecord
 
   has_one :filter, :dependent => :destroy
 
-  accepts_nested_attributes_for :filter, allow_destroy: true
+  accepts_nested_attributes_for :filter
```
Comment (ShimShtein) [inline app/models/endpoint.rb]: Why should we restrict existence of endpoints without filters? It doesn't interfere with the process, but can make user's life easier.
If you are worried about the user not understanding that this endpoint will not be hit, we can disable it automatically, or just show the fact in status.

Comment (adamruzicka) [reply app/models/endpoint.rb]: Functionally there is no difference between an endpoint with filter with no apps/event_types/levels and an endpoint without a filter. If an endpoint has no filter and the user updates the endpoint with apps/event_types/levels a filter gets automagically created. At this point, the user doesn't need to know that there is some kind of filter object, let alone be allowed to destroy it.

Comment (ShimShtein) [reply app/models/endpoint.rb]: filter with no apps/event_types/levels is allow all. Please make sure it's clear to the user.
We don't have a "deny all" notion in filters.

Comment (adamruzicka) [reply app/models/endpoint.rb]: Ok, I forgot about this. Rolled that change back

File: app/controllers/filters_controller.rb [hunk 2]
```diff
@@ -1,79 +1,15 @@
...
-    filter
-  end
-
-  def find_filter
-    find_params = {}
-    find_params[:endpoint_id] = params[:endpoint_id] if params[:endpoint_id]
-    find_params[:id] = params[:id] if params[:id]
-    @filter = authorize Filter.find_by(find_params)
+  def find_endpoint
+    @endpoint = authorize Endpoint.find(params.require(:endpoint_id))
```
Comment (ShimShtein) [inline app/controllers/filters_controller.rb]: Why do you need to query the endpoint? You can use `Filter.find_by(endpoint_id: params.require(:endpoint_id))`

File: spec/controllers/filters_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      relationships = data['relationships']
+      expect(relationships['apps']['data']).to eq([])
+      expect(relationships['event_types']['data']).to eq([])
+      expect(relationships['levels']['data']).to eq([])
+      endpoint_data = relationships['endpoint']['data']
+      expect(endpoint_data['id']).to eq(endpoint.id.to_s)
+      expect(endpoint_data['type']).to eq('endpoint')
+    end
+
+    it 'succeeds if the endpoint exists but does not have an endpoint' do
```
Comment (ShimShtein) [inline spec/controllers/filters_controller_spec.rb]: Do you actually need it to fail silently? Why not returning `404`, it's more restful way?

File: app/controllers/filters_controller.rb (hunk 2 above)
Comment (adamruzicka) [reply app/controllers/filters_controller.rb]: Because I don't :)

File: spec/controllers/filters_controller_spec.rb (hunk 3 above)
Comment (adamruzicka) [reply spec/controllers/filters_controller_spec.rb]: The frontend was expecting that. I'll open a PR to the frontend repo as well

**Comment by ShimShtein** [comment]
Merged. Thanks @adamruzicka!

**Pull request merged**